### PR TITLE
🧪 [Test] Add unit tests for ResourceDownloadCoordinator

### PR DESCRIPTION
**Pull request by dogi**
🎯 **What:** The testing gap addressed
This pull request adds unit tests for `startBackgroundDownload` in `ResourceDownloadCoordinator`, which previously lacked test coverage. 

📊 **Coverage:** What scenarios are now tested
The unit tests verify the background download initiation logic, including:
- **Happy Path:** When the server is available and URLs are not empty, verify that `DownloadUtils.openDownloadService` is called.
- **Edge Cases:**
  - When the server is unavailable, verify the service is not started.
  - When the URL list is empty, verify the service is not started.

✨ **Result:** The improvement in test coverage
The code logic is now comprehensively covered by tests leveraging MockK for simulating dependencies (`ConfigurationsRepository`, `DownloadUtils`) and `kotlinx-coroutines-test` for testing the suspended launch behavior, ensuring no regressions to future changes.

---
*PR created automatically by Jules for task [6124530464924229193](https://jules.google.com/task/6124530464924229193) started by @dogi*

### CODE REVIEW

**Review by Okuro3499 — DISMISSED** [review]
```
Justify the rating with specific references to the changes.                                                                                                                        
                                                                                                                                                                                     
  Assess relevance:                                                                                                                                                                  
                                                                                                                                                                                     
  What problem does this solve for the app?                                                                                                                                          
                                                                                                                                                                                     
  Is this change necessary now, or is it premature/over-engineered?                                                                                                                  
                                                                                                                                                                                     
  Call out risks (performance, data integrity, security, maintainability).                                                                                                           
                                                                                                                                                                                     
  List concrete recommendations (must-fix vs nice-to-have)                                                                                                                           
                                                          
⏺ I'll look at the diff on this branch before giving a verdict.                                                                                                                      
                                                                                                                                                                                   
⏺ Bash(git diff master...HEAD --stat)                                                                                                                                                
  ⎿   .../services/ResourceDownloadCoordinator.kt        |  2 +
      .../services/ResourceDownloadCoordinatorTest.kt    | 87 ++++++++++++++++++++++                                                                                                 
      2 files changed, 89 insertions(+)                                                                                                                                            
                                                                                                                                                                                     
⏺ Bash(git diff master...HEAD)                                                                                                                                                       
  ⎿  diff --git a/app/src/main/java/org/ole/planet/myplanet/services/ResourceDownloadCoordinator.kt b/app/src/main/java/org/ole/planet/myplanet/services/ResourceDownloadCoordin   
     ator.kt                                                                                                                                                                         
     index 74fe79de5..cead57143 100644                                                                                                                                             
     … +43 lines (ctrl+o to expand)                                                                                                                                                  
                                                                                                                                                                                     
⏺ Now let me read the full production file for context.                                                                                                                            
                                                                                                                                                                                     
  Searched for 2 patterns, read 1 file (ctrl+o to expand)                                                                                                                            
                                                                                                                                                                                   
⏺ ---                                                                                                                                                                                
  Branch Summary                                                                                                                                                                   
                                                                                                                                                                                     
  Adds unit tests for ResourceDownloadCoordinator.startBackgroundDownload, covering all three execution branches (happy path, server unavailable, empty URL list), plus a single     
  explanatory comment on the production method.                                                                                                                                      
                                                                                                                                                                                     
  ---                                                                                                                                                                                
  Rating: 2 — Needs Improvement                                                                                                                                                      
                                                                                                                                                                                     
  The direction is right — ResourceDownloadCoordinator had no tests, the logic has three distinct branches, and all three are now exercised. But there are concrete issues that
  should be fixed before merging.                                                                                                                                                    
                                                                                                                                                                                   
  ---                                                                                                                                                                                
  Assessment                                                                                                                                                                       

  What problem does this solve?
  ResourceDownloadCoordinator is a singleton that gates offline resource downloads behind server availability. It had zero test coverage. The tests verify the two guard conditions
  (checkServerAvailability and urls.isNotEmpty()) behave as expected. That is a real gap worth closing.                                                                              
   
  Is it premature?                                                                                                                                                                   
  No. The class is already in production use. Adding tests now is appropriate, not speculative.                                                                                    
                                                                                                                                                                                     
  ---                                                                                                                                                                              
  Risks & Issues                                                                                                                                                                     
                                                                                                                                                                                   
  Must-fix
                                                                                                                                                                                     
  1. Gratuitous comment in production code (ResourceDownloadCoordinator.kt:19)                                                                                                       
                                                                                                                                                                                     
  // Starts a background download if the server is available                                                                                                                         
  fun startBackgroundDownload(urls: ArrayList<String>) {                                                                                                                           

  This comment restates the function name and violates the project's own CLAUDE.md convention: "Only add [a comment] when the WHY is non-obvious… Don't explain WHAT the code does,  
  since well-named identifiers already do that." Remove it.
                                                                                                                                                                                     
  2. MainApplication.applicationScope mutated in setup() but never restored in tearDown()                                                                                            
   
  MainApplication.applicationScope = TestScope(testDispatcher)                                                                                                                       
                                                                                                                                                                                   
  applicationScope is a companion object lateinit var. Overwriting it without resetting in tearDown means if any other test class in the same JVM process reads                      
  MainApplication.applicationScope, it gets the stale TestScope. This is a latent test-pollution bug. Add a cleanup — either save and restore the original reference, or at minimum
  cancel the TestScope in tearDown.                                                                                                                                                  
                                                                                                                                                                                   
  // tearDown addition
  (MainApplication.applicationScope as? TestScope)?.cancel()
```

File: app/src/test/java/org/ole/planet/myplanet/services/ResourceDownloadCoordinatorTest.kt.orig [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import io.mockk.unmockkAll
+import io.mockk.verify
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import kotlinx.coroutines.test.TestScope
+import org.junit.After
```
Comment (Okuro3499) [inline app/src/test/java/org/ole/planet/myplanet/services/ResourceDownloadCoordinatorTest.kt.orig:20]: this is not testable

**Review by Okuro3499 — CHANGES_REQUESTED** [review]
